src/azure-cli/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -61,29 +172,81 @@ def load_arguments(self, _):    # pylint: disable=too-many-statements
         c.argument('tags', arg_type=tags_type)
         c.argument('included_event_types', arg_type=included_event_types_type)
         c.argument('labels', arg_type=labels_type)
-        c.argument('endpoint_type', arg_type=get_enum_type(['webhook', 'eventhub', 'storagequeue', 'hybridconnection', 'servicebusqueue'], default='webhook'))
+        c.argument('endpoint_type', arg_type=get_enum_type(['webhook', 'eventhub', 'storagequeue', 'hybridconnection', 'servicebusqueue', 'servicebustopic', 'azurefunction'], default='webhook'))
+        c.argument('delivery_identity_endpoint_type', arg_type=get_enum_type(['webhook', 'eventhub', 'storagequeue', 'hybridconnection', 'servicebusqueue', 'servicebustopic', 'azurefunction'], default=None), is_preview=True)
         c.argument('source_resource_id', help="Fully qualified identifier of the source Azure resource.")
-        c.argument('resource_id', deprecate_info=c.deprecate(redirect="--source-resource-id", expiration='3.0.0', hide=True), help="Fully qualified identifier of the Azure resource.")
-        c.argument('endpoint', help="Endpoint where EventGrid should deliver events matching this event subscription. For webhook endpoint type, this should be the corresponding webhook URL. For other endpoint types, this should be the Azure resource identifier of the endpoint. It is expected that the destination endpoint to be already created and available for use before executing any Event Grid command.")
+        c.argument('endpoint', help="Endpoint where EventGrid should deliver events matching this event subscription. For webhook endpoint type, this should be the corresponding webhook URL. For other endpoint types, this should be the Azure resource identifier of the endpoint.")
+        c.argument('delivery_identity_endpoint', help="Endpoint with identity where EventGrid should deliver events matching this event subscription. For webhook endpoint type, this should be the corresponding webhook URL. For other endpoint types, this should be the Azure resource identifier of the endpoint.", is_preview=True)
```

Comment:
through the code change, it seems that you change the argument name here, right? If yes, the change is breaking and will crash existing command.

Reply:
Which argument are you referring to? resource_id argument is deprecated for some time now and it was in effect for some time now. We are not introducing any breaking change here since resource_id is not in effect and passed the deprecation point.